Automattic/jetpack
# Social Menu & Social Media Icons: add StackOverflow support

I recently wanted to add my StackOverflow profile to my website but I found out that jetpack didn't have support for it.

So I decided to add the icons myself.

You can see it working here:
https://alazierplace.com/

I've also added support for [dev.to](https://dev.to/) and can make a PR for it if requried.

Furthermore, I read that there should be some general functionality i.e. the user should be able to add a custom icon that isn't supported. I'm willing to take that on as well if someone can point me in the right direction.

Regards.


#### Testing instructions:
1. Go to `Appearance -> Widgets`
2. Add `Social Icons (Jetpack)` widget to a widget area.
3. Add a stackoverflow.com link to the widget (e.g. https://stackoverflow.com/users/story/4678750)
4. The icon should render on the front-end with stackoverflow's logo icon. 

![image](https://user-images.githubusercontent.com/14261053/53297147-89584480-383b-11e9-963e-bf3e460c084d.png)


#### Proposed changelog entry for your changes:
* StackOverflow is now supported in Social Icons widget.
